exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/ProtoSerializer.java

```diff
@@ -140,6 +151,18 @@ protected void writeEndRepeatedPrimitive() {
     // Do nothing
   }
 
+  @Override
+  protected void writeStartRepeatedVarint(ProtoFieldInfo field, int payloadSize)
```

Comment:
The `ProtoSerializer` implementation needs the payload size to write repeated varints ([docs on that](https://developers.google.com/protocol-buffers/docs/encoding#packed)). It means the the size gets calculated twice for the payload, in two different places. Once when calculating overall size in the marshaler, and again when serializing. I don't think it's a huge deal but thought I'd call it out.

Reply:
Sorry for the late reply, I've been off for the holiday period! 

Basically with the current flow, it might be possible but from what I can see it would be awkward and would have to somewhat deviate from the current pattern. It's mainly because the `calculateSize` is `private static` and the result is passed into `super`: https://github.com/open-telemetry/opentelemetry-java/pull/3995/files#diff-23872c8146e426d9040de8d826fd7fa28f7473da4384de0ab5c6b002701186d9R26

So to me there isn't somewhere where I can stop and set an instance field, since the size is passed to `super` which has to be done first. @anuraaga might have an idea of a clean way to memoize it?

Reply:
Let's stick with this for now, thanks for looking into it